polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -229,31 +234,74 @@ pub fn worker_entrypoint(
 					},
 				};
 
-				// SAFETY: new process is spawned within a single threaded process. This invariant
-				// is enforced by tests.
-				let result = match unsafe { nix::unistd::fork() } {
-					Err(errno) => Err(error_from_errno("fork", errno)),
-					Ok(ForkResult::Child) => {
-						// Dropping the stream closes the underlying socket. We want to make sure
-						// that the sandboxed child can't get any kind of information from the
-						// outside world. The only IPC it should be able to do is sending its
-						// response over the pipe.
-						drop(stream);
-						// Drop the read end so we don't have too many FDs open.
-						drop(pipe_reader);
-
-						handle_child_process(
-							pvf,
-							pipe_writer,
-							preparation_timeout,
-							prepare_job_kind,
-							executor_params,
-						)
-					},
-					Ok(ForkResult::Parent { child }) => {
-						// the read end will wait until all write ends have been closed,
-						// this drop is necessary to avoid deadlock
-						drop(pipe_writer);
+				cfg_if::cfg_if! {
+					if #[cfg(any(target_os = "linux"))] {
```

Comment:
Nice usage of `cfg_if`! Do we need the `any` here?

Reply:
I don't think so. Done this way cause I was looking at how `nix` handled it. 
One unrelated point, I noticed that nix also has `pipe` in an already imported feature, what do you think about removing `os_pipe` and using the nix one to reduce dependency?

Reply:
> I don't think so. Done this way cause I was looking at how nix handled it.

Probably not needed in nix either. ;) Let's remove it.

> One unrelated point, I noticed that nix also has pipe in an already imported feature, what do you think about removing os_pipe and using the nix one to reduce dependency?

Great observation! I just looked at the source for both, and it looks like `os_pipe` does a bit more to be safer and cross-platform. It also returns separate types for the reader and writer, which is better for type safety, whereas nix only returns a pair of `OwnedFd`s.

`os_pipe` is a pretty small dependency though. If you want to remove one dependency from the codebase, it could be nice to build a wrapper over `nix`'s `pipe`, that does the stuff from `os_pipe`. I'd say it's not necessary though.